examples/cpp/core/bf/main.cpp

```diff
@@ -193,23 +193,28 @@ bool TranscodeFrom(Reader reader, const Options& options)
     }
 }
 
-
-bool Transcode(InputFile& input, const Options& options)
+template <typename Input>
+bool Transcode(Input input, const Options& options)
 {
-    switch (options.from)
+    bf::Protocol from = options.from.empty() ? guess : options.from.front();
+
+    if (from == guess)
+        std::cerr << std::endl << "Guessed " << ToString(from = Guess(input)) << std::endl;
```

Comment:
It wasn't changed from the previous version. But you are right, it is cleaner when separate.